builtin/logical/database/path_roles.go

```diff
@@ -713,13 +744,20 @@ func (b *databaseBackend) pathStaticRoleCreateUpdate(ctx context.Context, req *l
 }
 
 type roleEntry struct {
+	Name             string                 `json:"name"`
```

Comment:
At first glance it is not clear to me where we use this new field. Is it required for the Enterprise bits?

Reply:
Ah never mind, I see it makes it easier to store the roles using the helper `StoreStaticRole`. Is that correct? This field should have no impact on the user's workflow, just used internally by us to store the data

Reply:
> I see it makes it easier to store the roles using the helper StoreStaticRole

Exactly! Also, I have always found it odd that a `roleEntry` does not know its own name. This change should allow us to remove `RoleName` field from the `setStaticAccountInput` struct since its `Role` field holds the name now. However, I didn't want to make these changes too difficult to follow.

Reply:
Gotcha, that makes sense! Thanks for clarifying 🙏🏼